### PR TITLE
[#150015077] Fixing http auth and integrating it into ejabberd core

### PR DESCRIPTION
**Pull request by IRog**
In http_auth I had to modify the startup method. The supervision tree is slightly different but it shouldn't pose problems. I dunno why using the ejabberd_sup was causing issues, but I switched it to using direct start_links and there is still a supervisor in the chain, so it should be okay.

### CODE REVIEW

File: src/ejabberd_auth.erl [hunk 1]
```diff
@@ -524,61 +524,76 @@ db_set_password(User, Server, Password, Mod) ->
...
-	    ets_cache:lookup(?AUTH_CACHE, {User, Server});
-	false ->
-	    error;
-	true when UseCache ->
-	    ets_cache:lookup(
-	      ?AUTH_CACHE, {User, Server},
-	      fun() -> Mod:get_password(User, Server) end);
-	true ->
-	    Mod:get_password(User, Server)
+    case Mod == ejabberd_auth_http of
```
Comment (myorg) [inline src/ejabberd_auth.erl]: Why was this needed?  Just wondering.

Comment (myorg) [inline src/ejabberd_auth.erl]: Looks like you're wrapping the preexisting logic in a case check to make sure that the mod is for the AUTH module.  Is that correct?

Comment (IRog) [inline src/ejabberd_auth.erl]: Yep. There was no other good way to fork the flow. Sometimes the check done in the parent case would explode if evaluated, so I needed to pre-empt that.

Comment (myorg) [inline src/ejabberd_auth.erl]: Got it.  Finished looking at the rest.  LGTM.

Comment (IRog) [inline src/ejabberd_auth.erl]: Thanks for looking. I'll wait for @thePhilGuy then merge.

File: src/ejabberd_auth_http.erl [hunk 2]
```diff
@@ -0,0 +1,279 @@
+%%%----------------------------------------------------------------------
```
Comment (thePhilGuy) [inline src/ejabberd_auth_http.erl]: Did you change anything in the file below?

Comment (IRog) [inline src/ejabberd_auth_http.erl]: "In http_auth I had to modify the startup method."

**Comment by thePhilGuy** [comment]
🎩
